build_tools/cmake/iree_hal_cts_test_suite.cmake

```diff
@@ -84,10 +86,15 @@ function(iree_hal_cts_test_suite)
 
     set(_TRANSLATE_FLAGS
       "--compile-mode=hal-executable"
-      "--iree-hal-target-backends=${_RULE_COMPILER_TARGET_BACKEND}"
       ${_RULE_COMPILER_FLAGS}
     )
 
+    if(DEFINED _RULE_COMPILER_TARGET_DEVICE)
+      list(APPEND _TRANSLATE_FLAGS "--iree-hal-target-device=${_RULE_COMPILER_TARGET_DEVICE}")
+    else()
+      list(APPEND _TRANSLATE_FLAGS "--iree-hal-target-backends=${_RULE_COMPILER_TARGET_BACKEND}")
+    endif()
```

Comment:
> Adding COMPILER_TARGET_DEVICE to iree_hal_cts_test_suite.
COMPILER_TARGET_BACKEND is something we should deprecate in the future.

Agreed. I've seen `--iree-hal-target-device` usage popping up and tribal knowledge being passed around ("I heard someone say to use this instead of that") but the code and docs in this repository are still lagging behind.

Argh, time flies: https://github.com/iree-org/iree/issues/17977#issuecomment-2248413416